src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -1150,13 +1159,13 @@ void IMergeTreeDataPart::remove() const
       */
 
     fs::path from = fs::path(storage.relative_data_path) / relative_path;
-    fs::path to = fs::path(storage.relative_data_path) / ("delete_tmp_" + name);
+    fs::path to = fs::path(storage.relative_data_path) / ("delete_tmp_" + relative_path);
```

Comment:
The fix is here

Reply:
Maybe add a comment? So now we will have something like `delete_tmp_tmp_merge_<part_name>` or `delete_tmp_tmp_fetch_<part_name>` instead of `delete_tmp_<part_name>`?